addons/escoria-core/game/main.gd

```diff
@@ -40,9 +40,10 @@ func set_scene(p_scene: Node) -> void:
 	if current_scene != null:
 		clear_scene()
 
-	if not p_scene in get_children():
+	if not p_scene.is_inside_tree() and not p_scene in get_children():
```

Comment:
I'm wondering about the second clause here. If the scene is not in the tree, it will never be in the children of the current node, right?

Reply:
Indeed, that's an `or` check here, not an `and`